rust/hyperlane-core/src/traits/validator_announce.rs

```diff
@@ -0,0 +1,18 @@
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use auto_impl::auto_impl;
+
+use crate::{ChainResult, HyperlaneContract, H256};
+
+/// Interface for the ValidatorAnnounce chain contract. Allows abstraction over
+/// different chains
+#[async_trait]
+#[auto_impl(Box, Arc)]
+pub trait ValidatorAnnounce: HyperlaneContract + Send + Sync + Debug {
+    /// Returns the announced storage locations for the provided validators.
+    async fn get_announced_storage_locations(
+        &self,
+        validators: &[H256],
+    ) -> ChainResult<Vec<Vec<String>>>;
```

Comment:
Can we create a type for this so it looks more like `Vec<Validator>` or something. Seeing `Vec<Vec<String>>` leaves a lot up to interpretation. Also is `String` the most appropriate inner type?

Reply:
I guess I was thinking it should more or less match what is returned by the smart contract, and then to process those results at a higher layer.

Alternatively it could return `Vec<Vec<Option<CheckpointSyncerConf>>` and we could just call `CheckpoitnSyncerConf::from_storage_location()` on the results?

I dont' feel strongly